cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -166,6 +169,32 @@ func createDevice(device *v1beta1.Device) types.Device {
 	return edgeDevice
 }
 
+// isExistModel check if the target node already has the model.
+func isExistModel(deviceMap *sync.Map, device *v1beta1.Device) bool {
+	var res bool
+	targetNode := device.Spec.NodeName
+	modelName := device.Spec.DeviceModelRef.Name
+	// To find another device in deviceMap that uses the same deviceModel with exclude current device
+	deviceMap.Range(func(k, v interface{}) bool {
+		if k == device.Name {
+			return true
+		}
+		deviceItem, ok := v.(*v1beta1.Device)
+		if !ok {
+			return true
+		}
+		if deviceItem.Spec.NodeName == "" {
+			return true
+		}
+		if deviceItem.Spec.NodeName == targetNode && deviceItem.Spec.DeviceModelRef.Name == modelName {
```

Comment:
cc @WillardHu @wbc6080

Reply:
> Because the device info has been added to `deviceMap` at L125, so this range will always compare will itself, and set res to true? @cl2017

This logic seems to expect that the same device model will be reused at the same node, right? @cl2017 
So the res is not always true, it returns a false value when the device model first binds the node.

Reply:
In L179, If the currently index device in the map is the device just added to the map, we will directly skip the check of this index device. Therefore, this range will not be affected. Is it right? @cl2017

Reply:
> > Because the device info has been added to `deviceMap` at L125, so this range will always compare will itself, and set res to true? @cl2017
> 
> This logic seems to expect that the same device model will be reused at the same node, right? @cl2017 So the res is not always true, it returns a false value when the device model first binds the node.

Yes, this function is used to find wether the same `devicemodel` already exists at the `targetNode`.
At this fuction, we use `res` variable as a result, rather than the `return` value in `Range` method, `return true` means continue `Range` function.
@fisherxu @WillardHu @wbc6080

Reply:
> In L179, If the currently index device in the map is the device just added to the map, we will directly skip the check of this index device. Therefore, this range will not be affected. Is it right? @cl2017

Make sense, if device name == k, it means this is itself. Then return. Right? @wbc6080 

And another question, only compare name maybe not enough, it can be same with device in another namespace.. ptal @cl2017

Reply:
Well,this is indeed a problem. I also ignored that device instance and device model are namespace level resources.This may also cause another problem. If two namespaces have a model with the same name, the model in namespace B may interfere with the model referenced by the device in namespace A during range. @fisherxu @cl2017

Reply:
I think we can fix the device name in follow-up PRs, and could you please open an issue to track it? @wbc6080 @cl2017

Reply:
> I think we can fix the device name in follow-up PRs, and could you please open an issue to track it? @wbc6080 @cl2017

Yes, most device code has not considered the possible problems caused by different namespaces.
Let's open an issue, tracking and resolve these problems together.